addons/portal/controllers/portal.py

```diff
@@ -308,10 +308,65 @@ def account(self, redirect=None, **post):
     def on_account_update(self, values, partner):
         pass
 
+    def _get_commercial_partners_and_users(self):
+        partner = request.env.user.partner_id
+        commercial_partners = partner.commercial_partner_id.child_ids
+        portal_users = commercial_partners.filtered(lambda partner: partner.user_ids and partner.user_ids[0].has_group('base.group_portal')) \
+                                          .mapped(lambda partner: partner.user_ids[0] if partner.user_ids else partner.user_ids)
```

Comment:
could partner inside lambda conflict with partner declared at the beginning ?
maybe rename them `com_partner`